tests/specifications/polymorphic.yaml

```diff
@@ -37,8 +37,6 @@ components:
     ParentType:
       type: object
       properties:
-        value:
```

Comment:
Could you please definition in more detail why this description is no longer necessary?

Reply:
So in main generated code looks like this:
Notice multiple `...overrideResponse` and overrideResponse is of type `any`
```ts
export const getGetPolymorphicResponseResponseMock = (
  overrideResponse: any = {},
): GetPolymorphicResponse200 =>
  faker.helpers.arrayElement([
    {
      key: faker.helpers.arrayElement([faker.word.sample(), undefined]),
      type: faker.helpers.arrayElement([
        faker.helpers.arrayElement(['BOOLEAN', 'STRING'] as const),
        undefined,
      ]),
      value: faker.helpers.arrayElement([{}, undefined]),
      ...overrideResponse,
      ...overrideResponse,
    },
    {
      key: faker.helpers.arrayElement([faker.word.sample(), undefined]),
      type: faker.helpers.arrayElement([
        faker.helpers.arrayElement(['BOOLEAN', 'STRING'] as const),
        undefined,
      ]),
      value: faker.helpers.arrayElement([{}, undefined]),
      ...overrideResponse,
      ...overrideResponse,
    },
  ]);
```

as this PR is stoped overriding nested data in generated code same part looks like this:
```ts
export const getGetPolymorphicResponseResponseMock =
  (): GetPolymorphicResponse200 =>
    faker.helpers.arrayElement([
      {
        key: faker.helpers.arrayElement([faker.word.sample(), undefined]),
        type: faker.helpers.arrayElement([
          faker.helpers.arrayElement(['BOOLEAN', 'STRING'] as const),
          undefined,
        ]),
        value: faker.helpers.arrayElement([{}, undefined]),
      },
      {
        key: faker.helpers.arrayElement([faker.word.sample(), undefined]),
        type: faker.helpers.arrayElement([
          faker.helpers.arrayElement(['BOOLEAN', 'STRING'] as const),
          undefined,
        ]),
        value: faker.helpers.arrayElement([{}, undefined]),
      },
    ]);
```

as spread of `any` is gone TS starts to complain because `GetPolymorphicResponse200` is impossible type

here is a link to TS playground
https://www.typescriptlang.org/play/?#code/C4TwDgpgBACghgJwgO2AFXBDkoF4qiQD2AZrIiupthANoDWEIpBmL8SqNNAugNwAoAQGMiyAM7BynKpBp4oAbwFQoAIQDyGgDIBRAIIA5AFxQA5Jp0HDZgDQqoAZTQAlAJKGA4qbPP3XuwEAXyg4cShRCWBBAQgADzAiBCkAS1QIBBI4YWgOSnllVUYQAH5TSQQ0gHNBVUIIMul86kxaqAA3OAAbAFcG00UoBiZy4ErkKp5TOGQQKCDBIKF6qAARCHEc5AATGeANZGh8PK5MKAAyJQdO3v6oACMiIi6IGcXBeMTk1hx1zZRdlwAO5EBQnWTQS6FDrdPqNCrVd7LM6eCDAGDPEAAWySYAAFilhC4NokJBAAEwABkpCj+W0B+0OUAAPmsNvS9mgQTFIpIoGIIKZUejMTiEPjCcTxKTxBTqQpoUURuYBYFVHVMD5LHojGr1Tc+qYxn17EsRGI+cAQUK0RiutjcQSiSSLXKafhFVBij4rUQ9RrID4-B5PP6YbcfDM5giJoEzbypFGDkcruqvcqzKr7Gn6lqtDqbNn9bDBQQECaHKoAHQ1xQhMKhWamoA

I don't know at which point things got broken because looking at original `polymorphic.yaml` in swagger editor
`value` key should be ignored from `ParentType`

Would make sense to do a followup PR rather than try to fix things in this PR

Reply:
Thank you for letting me know so politely.
In other words, the mock was originally broken, and by expanding `overrideResponse`, the error was suppressed and it worked, right?

Reply:
One possible solution could be to type the different possibilities, like this:
```ts
faker.helpers.arrayElement([
  {
    key: faker.helpers.arrayElement([faker.word.sample(), undefined]),
    type: faker.helpers.arrayElement([
      faker.helpers.arrayElement(['BOOLEAN', 'STRING'] as const),
      undefined,
    ]),
    value: faker.helpers.arrayElement([{}, undefined]),
  } as DescendantOne,
  {
    key: faker.helpers.arrayElement([faker.word.sample(), undefined]),
    type: faker.helpers.arrayElement([
      faker.helpers.arrayElement(['BOOLEAN', 'STRING'] as const),
      undefined,
    ]),
    value: faker.helpers.arrayElement([{}, undefined]),
  } as DescendantTwo,
]);
```

One improvement here could maybe also be to add the "overridden" type to the array as well? Something like
```ts

faker.helpers.arrayElement([
  {
    key: faker.helpers.arrayElement([faker.word.sample(), undefined]),
    type: faker.helpers.arrayElement([
      faker.helpers.arrayElement(['BOOLEAN', 'STRING'] as const),
      undefined,
    ]),
    value: faker.helpers.arrayElement([{}, faker.datatype.boolean(), undefined]),
  } as DescendantOne,
  {
    key: faker.helpers.arrayElement([faker.word.sample(), undefined]),
    type: faker.helpers.arrayElement([
      faker.helpers.arrayElement(['BOOLEAN', 'STRING'] as const),
      undefined,
    ]),
    value: faker.helpers.arrayElement([{}, faker.word.sample(), undefined]),
  } as DescendantTwo,
]);
```

Reply:
feels like I opened can of worms 😅

Issue is not in faker code, it's `GetPolymorphicResponse200` that's broken.
that type cannot be used even outside of faker.

I don't mind to investigate the case but let's consider this

generated types are broken already.
as no-one reported broken types probably it means no-one has such use case in their project.

hence I would argue that we're not actually breaking anyones code.

I'm on vacation next week so any update from my side will arrive with delay if I don't dig up anything today.

Reply:
So I traveled back to commit where `polymorphic.yaml` was added

faker types were not typed at that time
```ts
export const getGetPolymorphicResponseMock = () =>
  faker.helpers.arrayElement([
    {
      value: faker.helpers.arrayElement([{}, undefined]),
      key: faker.helpers.arrayElement([faker.random.word(), undefined]),
      type: faker.helpers.arrayElement([
        faker.helpers.arrayElement(['BOOLEAN', 'STRING']),
        undefined,
      ]),
    },
    {
      value: faker.helpers.arrayElement([{}, undefined]),
      key: faker.helpers.arrayElement([faker.random.word(), undefined]),
      type: faker.helpers.arrayElement([
        faker.helpers.arrayElement(['BOOLEAN', 'STRING']),
        undefined,
      ]),
    },
  ]);
```

I suspect Orval never really supported situation that was created in `polymorphic.yaml`

According to Official Swagger [docs](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/#allof) Orval is generating correct types and `polymorphic.yaml` is broken.

There is no hierarchy in `allOf` hence data should comply with all schemas thus I don't think ignoring `value: Record` from `ParentType` is a way forward.

Reply:
@karlismelderis-mckinsey 

Thank you for investigating.
I understand that `Polymorphic` is not working and the types being generated are flawed, and the problem manifests itself not only in mocks. What I was concerned about was breaking a working state in an unintentional way.
However, even if we only take care of the mock with this PR, the fundamental problem will not be resolved and it is not essential, so we agree to treat it as a separate issue in favor of the original purpose of this PR.

Reply:
@soartec-lab generated types are not flawed.
Quite contrary, they show just what `polymorphic.yaml` intends to.

But `polymorphic.yaml` is flawed as a test example.
`value` key should not be part of `ParentType` schema

Hence if we agree to my change (removal of `value` from `ParentType`) I would consider test data fixed and case closed

Reply:
@karlismelderis-mckinsey 

I see, I understood that there is no problem with the generated type, but the problem is with the `OpenAPI` definition. If this change causes problems for some users, they will need to modify their own `OpenAPI` definitions.

I was able to understand correctly. And I think that's correct.